### PR TITLE
Merge Development->Master (led funcs, nanopb)

### PR DESCRIPTION
**Pull request by maartentamboer**
Add the led functions and nanopb to master branch

### CODE REVIEW

File: src/leds.c [hunk 1]
```diff
@@ -0,0 +1,73 @@
+#include "leds.h"
```
Comment (xor-gate) [inline src/leds.c:1]: better prefix all functions with led_* or better `oca_led_*` then you know the scope.

File: src/leds.c [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+#include "ch.h"
+#include "hal.h"
+#include "board.h"
+
+enum LedState actLed;
+enum LedState statLed;
+enum LedState errLed;
```
Comment (xor-gate) [inline src/leds.c]: LedState is no new type, dont do typedefs (see linux codestyle). Better would be to place the leds states in a struct prefix with `g_*` and make it static.

File: src/leds.c [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+}
+
+void toggle_led(enum LedSelect led)
+{
+	palTogglePad(GPIOB, led);
+}
+
+void update_leds(void)
+{
+	if(actLed == led_blink_250){ toggle_led(led_act); }
```
Comment (xor-gate) [inline src/leds.c]: the linux checkpatch linter will come after you :+) single line is enough without braces and should be placed on the next line

**Review by xor-gate — CHANGES_REQUESTED** [review]
Few remarks, looks good.

**Comment by maartentamboer** [comment]
Review again? Should be more in line with coding style now

File: include/leds.h [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+	oca_led_blink_500,
+	oca_led_blink_1000
+};
+
+void oca_init_leds(void);
+
+void oca_set_led(enum oca_led_select led, enum oca_led_state state);
+void oca_toggle_led(enum oca_led_select led);
+
+void oca_update_leds(void); //call every 250ms
```
Comment (xor-gate) [inline include/leds.h]: almost, in C programming you could fake namespaces as in C++ classes with `<project>_<file>_<function>`. So for oca led update write: 
`oca_led_update` or `oca_led_toggle`. This is easy when you mixup alot of code from other projects and places. Let me know what you think.

**Review by xor-gate — CHANGES_REQUESTED** [review]

**Comment by maartentamboer** [comment]
Changed

**Comment by xor-gate** [comment]
Great!

**Pull request merged**